client/container_create.go

```diff
@@ -108,3 +113,42 @@ func hasEndpointSpecificMacAddress(networkingConfig *network.NetworkingConfig) b
 	}
 	return false
 }
+
+// allCapabilities is a magic value for "all capabilities"
+const allCapabilities = "ALL"
+
+// normalizeCapabilities normalizes capabilities to their canonical form,
+// removes duplicates, and sorts the results.
+//
+// It is similar to [github.com/docker/docker/oci/caps.NormalizeLegacyCapabilities],
+// but performs no validation based on supported capabilities.
+func normalizeCapabilities(caps []string) []string {
+	var normalized []string
+
+	unique := make(map[string]struct{})
+	for _, c := range caps {
+		c = normalizeCap(c)
```

Comment:
Yeah, technically it would probably be fine to only keep `ALL`, but there's logic in the daemon to handle the `ALL` special case, and to hydrate the list of capabilities; see  `TweakCapabilities` https://github.com/moby/moby/blob/4001d0704ba38a82e1dbc26f0593fca66db1cb98/oci/caps/utils.go#L83

I tried to avoid (at least for now) putting _too much_ logic on the client side (other than normalising the format) and to preserve information where possible; also having situations in the back of my mind where a user could pass a capability in the list that's not (yet) known by the engine, and I'd want it to produce an error in that case instead of silently taking `ALL`.